### PR TITLE
Processor for Pylint

### PR DESCRIPTION
**Pull request by rafalnowak**
I've created new processor for running Pylint - neat tool for Python code analysis - https://www.pylint.org/

In one of the commits I have also refactored TSLint processor because it had some common parts with Pylint processor - both are based on running external process and parsing its output in some way, so I decided to make base type above them in order to avoid code duplication.

### CODE REVIEW

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import pl.touk.sputnik.exec.ExternalProcess;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@Slf4j
+public class PylintExecutor {
+    private static final String pylintExecutable = "pylint";
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: [Checkstyle] ERROR: Name 'pylintExecutable' must match pattern '^(log|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*)$'.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String pylintOutputFormat = "--output-format=json";
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: [Checkstyle] ERROR: Name 'pylintOutputFormat' must match pattern '^(log|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*)$'.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String pylintRcfileName = "--rcfile=";
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: [Checkstyle] ERROR: Name 'pylintRcfileName' must match pattern '^(log|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*)$'.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    private String removeHeaderFromPylintOutput(String pylintOutput) {
+        return pylintOutput.replace("No config file found, using default configuration", "");
+    }
+
+    private String formatViolationMessageFromPylint(PylintMessage message) {
+        return message.getMessage() + " [" + message.getSymbol() + "]";
+    }
+
+    private Severity pylintMessageTypeToSeverity(String messageType) {
+        if (messageType.equals("error") || messageType.equals("fatal")) {
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java]: [Checkstyle] ERROR: String literal expressions should be on the left side of an equals comparison.

Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java]: [Checkstyle] ERROR: String literal expressions should be on the left side of an equals comparison.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            return Severity.ERROR;
+        } else if (messageType.equals("warning")) {
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java]: [Checkstyle] ERROR: String literal expressions should be on the left side of an equals comparison.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            return Severity.WARNING;
+        } else if (messageType.equals("convention") || messageType.equals("refactor")) {
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java]: [Checkstyle] ERROR: String literal expressions should be on the left side of an equals comparison.

Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java]: [Checkstyle] ERROR: String literal expressions should be on the left side of an equals comparison.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java (hunk 1 above)
Comment (SpOOnman) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: +1

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java (hunk 2 above)
Comment (SpOOnman) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: +1

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java (hunk 3 above)
Comment (SpOOnman) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: +1

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        List<String> rcfileNameArg = getRcfileNameAsList();
+        List<String> filePathArg = ImmutableList.of(filePath);
+        List<String> allArgs = Lists.newArrayList(Iterables.concat(basicPylintArgs, rcfileNameArg, filePathArg));
+        return allArgs.toArray(new String[allArgs.size()]);
+    }
+
+    private List<String> getRcfileNameAsList() {
+        if (rcfileName != null) {
+            return ImmutableList.of(pylintRcfileName + rcfileName);
+        } else {
```
Comment (SpOOnman) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintExecutor.java]: This else is not needed.

File: src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java (hunk 6 above)
Comment (SpOOnman) [inline src/main/java/pl/touk/sputnik/processor/pylint/PylintResultParser.java]: All these strings should be constants

**Comment by SpOOnman** [comment]
Seems fine, only few contstants to extract.

**Comment by SpOOnman** [comment]
Thank you!

**Pull request merged**